tests/conftest.py

```diff
@@ -0,0 +1,70 @@
+import ctypes.util
+import os
+import pathlib
+import re
+
+import pytest
+
+# If you want to test pymagic using specific libmagic version:
+# - put `libmagic.so` and `magic.mgc` in `tests/libmagic` directory
+# - call LIBMAGIC_PRELOAD=FILE5_38 pytest if 5.38 is expected libmagic version:
+
+tests_dir = pathlib.Path(__file__).parent
+expected_libmagic = os.environ.get("LIBMAGIC_PRELOAD")
+
+if expected_libmagic:
+    version_match = re.match(r"FILE(\d)_(\d\d)", expected_libmagic)
+    if not version_match:
+        raise RuntimeError("LIBMAGIC_PRELOAD value doesn't match FILEx_xx format")
+    expected_version = int("".join(version_match.groups()))
+
+    libmagic_file = tests_dir / "libmagic" / "libmagic.so"
+    database_file = tests_dir / "libmagic" / "magic.mgc"
+    if not libmagic_file.exists() or not database_file.exists():
+        raise RuntimeError("LIBMAGIC_PRELOAD is set, but libmagic binaries are missing")
+
+    ctypes_find_library = ctypes.util.find_library
+
+    def find_library_patch(name):
+        if name == "magic":
+            return str(libmagic_file)
+        return ctypes_find_library(name)
+
+    ctypes.util.find_library = find_library_patch
+
+    import magic as pymagic
```

Comment:
I think that it would be useful to add a comment explaining why we're doing this.

Reply:
:+1: